code/parse/sexp.cpp

```diff
@@ -23109,13 +23110,13 @@ int eval_sexp(int cur_node, int referenced_node)
 	}
 
 	// now do a quick return whether or not we log, per the comment above about trapping known sexpressions
-	if (Sexp_nodes[cur_node].value == SEXP_KNOWN_TRUE) {
+	if (Sexp_nodes[cur_node].value == SEXP_KNOWN_TRUE && !(Sexp_nodes[cur_node].flags & SNF_SPECIAL_ARG_IN_NODE)) {
```

Comment:
I'm still the rookie but, we could nest these three conditionals inside a check for !(Sexp_nodes[cur_node].flags & SNF_SPECIAL_ARG, since they all check the same thing.

Reply:
Good shout, I really didn't like how ungainly those conditionals were. PR updated.